From d52625a94a816fffb880b8a85f3bfd4c35d02b99 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Tue, 27 Oct 2015 09:28:10 +0800
Subject: [PATCH] gwin32.c: Fix g_win32_check_windows_version() on 32-bit

The Windows API function RtlGetVersion() is actually a function that is
decorated by WINAPI (i.e. __stdcall), so we need to correct this so that
the symbol can be loaded correctly from ntdll.dll, so that we won't crash as
a result.  Should fix the crash due to stack overflow on 32-bit builds.

https://bugzilla.gnome.org/show_bug.cgi?id=756179
---
 glib/gwin32.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gwin32.c b/glib/gwin32.c
index ce462bb..4c89651 100644
--- a/glib/gwin32.c
+++ b/glib/gwin32.c
@@ -567,7 +567,7 @@ g_win32_check_windows_version (const gint major,
 
 #if WINAPI_FAMILY != MODERN_API_FAMILY
   /* For non-modern UI Apps, use the LoadLibraryW()/GetProcAddress() thing */
-  typedef NTSTATUS fRtlGetVersion (PRTL_OSVERSIONINFOEXW);
+  typedef NTSTATUS (WINAPI fRtlGetVersion) (PRTL_OSVERSIONINFOEXW);
 
   fRtlGetVersion *RtlGetVersion;
   HMODULE hmodule;
-- 
2.5.0

